src/main/java/net/starschema/clouddb/jdbc/BQSupportFuncts.java

```diff
@@ -364,14 +365,19 @@ public static String logAndGetQueryState(Job pollJob) throws IOException {
         || pollJob.getStatistics().isEmpty()) {
       throw new IOException("Failed to fetch query state.");
     }
-    BQSupportFuncts.logger.info(
-        "Job status: "
-            + pollJob.getStatus().getState()
-            + " ; "
-            + pollJob.getJobReference().getJobId()
-            + " ; "
-            + (System.currentTimeMillis() - pollJob.getStatistics().getStartTime()));
-    return pollJob.getStatus().getState();
+
+    if (Optional.ofNullable(pollJob.getStatistics().getCreationTime()).isPresent()) {
```

Comment:
nit: just check if it's null if the optional is never used again.

Reply:
On second thought don't even nest inside if ... try to avoid deep nesting
```
long creationTime = Optional.ofNullable(pollJob.getStatistics().getCreationTime()).getOrElse(pollJob.getStatistics().getStartTime())
```

is yet another option.